code/__HELPERS/paths/path.dm

```diff
@@ -0,0 +1,379 @@
+/**
+ * This is the proc you use whenever you want to have pathfinding more complex than "try stepping towards the thing".
+ * If no path was found, returns an empty list, which is important for bots like medibots who expect an empty list rather than nothing.
+ * It will yield until a path is returned, using magic
+ *
+ * Arguments:
+ * * caller: The movable atom that's trying to find the path
+ * * end: What we're trying to path to. It doesn't matter if this is a turf or some other atom, we're gonna just path to the turf it's on anyway
+ * * max_distance: The maximum number of steps we can take in a given path to search (default: 30, 0 = infinite)
+ * * mintargetdistance: Minimum distance to the target before path returns, could be used to get near a target, but not right to it - for an AI mob with a gun, for example.
+ * * access: A list representing what access we have and what doors we can open.
+ * * simulated_only: Whether we consider tur fs without atmos simulation (AKA do we want to ignore space)
+ * * exclude: If we want to avoid a specific turf, like if we're a mulebot who already got blocked by some turf
+ * * skip_first: Whether or not to delete the first item in the path. This would be done because the first item is the starting tile, which can break movement for some creatures.
+ * * diagonal_handling: defines how we handle diagonal moves. see __DEFINES/path.dm
+ */
+/proc/get_path_to(atom/movable/caller, atom/end, max_distance = 30, mintargetdist, access=list(), simulated_only = TRUE, turf/exclude, skip_first=TRUE, diagonal_handling=DIAGONAL_REMOVE_CLUNKY)
+	var/list/hand_around = list()
+	// We're guarenteed that list will be the first list in pathfinding_finished's argset because of how callback handles the arguments list
+	var/datum/callback/await = list(CALLBACK(GLOBAL_PROC, /proc/pathfinding_finished, hand_around))
```

Comment:
use `GLOBAL_PROC_REF()` instead in case lummox trolls us again with that

```suggestion
	var/datum/callback/await = list(CALLBACK(GLOBAL_PROC, GLOBAL_PROC_REF(pathfinding_finished), hand_around))
```